ginac/inifcns_trig.cpp

```diff
@@ -722,6 +776,18 @@ static ex cot_evalf(const ex & x, PyObject* parent)
 
 static ex cot_eval(const ex & x)
 {
+        std::function<bool (const ex&)> has_pi =
```

Comment:
I think we should no longer use several identical lambdas but one static function. Lambdas are also slower I heard. Could you please do a separate PR with thzis change?